lib/src/flutterflow_api_client.dart

```diff
@@ -265,3 +279,95 @@ Future _runFix({
     stderr.write('Error running "dart fix": $e\n');
   }
 }
+
+Future firebaseDeploy({
+  required String token,
+  required String projectId,
+  bool appendRules = false,
+  String endpoint = kDefaultEndpoint,
+}) async {
+  final endpointUrl = Uri.parse(endpoint);
+  final body = jsonEncode({
+    'project': {'path': 'projects/$projectId'},
+    'append_rules': appendRules,
+  });
+  final result = await _callEndpoint(
+    client: http.Client(),
+    token: token,
+    url: Uri.https(
+        endpointUrl.host, '${endpointUrl.path}/exportFirebaseDeployCode'),
+    body: body,
+  );
+
+  // Download actual code
+  final projectZipBytes = base64Decode(result['firebase_zip']);
+  final firebaseProjectId = result['firebase_project_id'];
+  final tmpFolder =
+      Directory.systemTemp.createTempSync('${projectId}_$firebaseProjectId');
```

Comment:
Nit: move this line to right above the `try`. Currently if it fails during, say, `extractArchiveToCurrentDirectory` (unlikely but not impossible) it will fail to clean up the temp directory.